File: 剑指offer/C++/MergeList.cpp
#include <iostream>


struct ListNode {
	int val;
	struct ListNode *next;
	ListNode(int x) :
			val(x), next(NULL) {
	}
};

class Solution
{
public:
    ListNode *Merge(ListNode *pHead1, ListNode *pHead2)
    {
        ListNode *res = new ListNode(0);
        ListNode *head = res;
        while (pHead1 != NULL && pHead2 != NULL)
        {
            if (pHead1->val <= pHead2->val)
            {
                res->next = new ListNode(pHead1->val);
                pHead1 = pHead1->next;
            } else {
                res->next = new ListNode(pHead2->val);
                pHead2 = pHead2->next;
            }
            res=res->next;
        }
        while (pHead1 != NULL)
        {
            res->next=new ListNode(pHead1->val);
            res=res->next;
            pHead1=pHead1->next;
        }
        while (pHead2 != NULL)
        {
            res->next = new ListNode(pHead2->val);
            res=res->next;
            pHead2=pHead2->next;
        }
        return head->next;
    }
};

int main(int argc,char **argv,char **envp){
    return 0;
}

File: 剑指offer/Java/MergeList.java
class ListNode {
    int val;
    ListNode next = null;

    ListNode(int val) {
        this.val = val;
    }
}
class Solution {
    public ListNode Merge(ListNode list1, ListNode list2) {
        ListNode res = new ListNode(0);
        ListNode h = res;
        while(list1!=null && list2!=null){
            if(list1.val<=list2.val){
                res.next = new ListNode(list1.val);
                list1 = list1.next;
            } else {
                res.next = new ListNode(list2.val);
                list2 = list2.next;
            }
            res = res.next;
        }
        while(list1 != null){
            res.next = new ListNode(list1.val);
            list1 = list1.next;
            res=res.next;
        }
        while(list2 != null){
            res.next = new ListNode(list2.val);
            list2 = list2.next;
            res = res.next;
        }
        return h.next;
    }
}

public class MergeList{
    public static void main(String[] args) {
        
    }
}